### PR TITLE
Dialog node update :

### PR DESCRIPTION
**Pull request by ylecleach**
- creation Mode now updated : file or Buffer in the msg.payload
- remove double quotes by simple quotes
- removed undesirable successful node status for all modes
- node inline documentation updated.

### CODE REVIEW

File: services/dialog/v1.js [hunk 1]
```diff
@@ -146,31 +146,31 @@ module.exports = function (RED) {
...
                 node.error(err, msg);
               } else {
-                node.status({fill:"green", shape:"dot", text:"dialog conversation successful"});		  
                 msg.dialog = dialog_data;		  
-                msg.payload = "Check msg.dialog dialog data";
+                msg.payload = 'Check msg.dialog dialog data';
                 node.send(msg);
               }   
             });
           }
```
Comment (yrezgui) [inline services/dialog/v1.js:163]: Codacy Issue found: [Closing curly brace does not appear on the same line as the subsequent block.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l164)

File: services/dialog/v1.js [hunk 2]
```diff
@@ -183,113 +183,114 @@ module.exports = function (RED) {
...
 
 // This function delete an existing dialog instance.
   function performDelete(node,dialog,msg, config) {
     var params = {};
-    var message = "";
-    node.status({fill:"blue", shape:"dot", text:"requesting delete of a dialog instance"}); 
+    var message = '';
+    node.status({fill:'blue', shape:'dot', text:'requesting delete of a dialog instance'}); 
 
-    if (!config.dialog || "" != config.dialog) {
```
Comment (yrezgui) [inline services/dialog/v1.js:232]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l235)

File: services/dialog/v1.js [hunk 3]
```diff
@@ -86,57 +86,57 @@ module.exports = function (RED) {
...
-              node.status({fill:"red", shape:"dot", text:message});	
+		    if (!clientid || '' === clientid) {
+              message = 'Missing Client ID';
+              node.status({fill:'red', shape:'dot', text:message});	
               node.error(message, msg);	
             }
-            if (!converseid || "" === converseid) {
-              var message = "Missing Conversation ID";
-              node.status({fill:"red", shape:"dot", text:message});	
+            if (!converseid || '' === converseid) {
```
Comment (yrezgui) [inline services/dialog/v1.js:137]: Codacy Issue found: ["message" is already defined](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l138)

File: services/dialog/v1.js [hunk 4]
```diff
@@ -146,31 +146,31 @@ module.exports = function (RED) {
...
-          if (config.mode === "startconverse" || config.mode === "converse") {			  
-            node.status({fill:"blue", shape:"dot", text:"Starting Dialog Conversation"});
+          if (config.mode === 'startconverse' || config.mode === 'converse') {			  
+            node.status({fill:'blue', shape:'dot', text:'Starting Dialog Conversation'});
             dialog.conversation (params, function (err, dialog_data) {
+              node.status({});
               if (err) {
-                node.status({fill:"red", shape:"ring", text:"call to dialog service failed"}); 
-                console.log("Error:", msg, err);
+                node.status({fill:'red', shape:'ring', text:'call to dialog service failed'});
```
Comment (yrezgui) [inline services/dialog/v1.js:154]: Codacy Issue found: [Unexpected console statement.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l155)

File: services/dialog/v1.js [hunk 5]
```diff
@@ -146,31 +146,31 @@ module.exports = function (RED) {
           params.dialog_id = dialogid;
```
Comment (yrezgui) [inline services/dialog/v1.js:146]: Codacy Issue found: [Expected indentation of 8 space characters but found 10.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l147)

File: services/dialog/v1.js (hunk 3 above)
Comment (yrezgui) [inline services/dialog/v1.js:137]: Codacy Issue found: [Expected blank line after variable declarations.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l138)

File: services/dialog/v1.js [hunk 6]
```diff
@@ -146,31 +146,31 @@ module.exports = function (RED) {
...
           params.input = msg.payload;
 		  
-          if (config.mode === "startconverse" || config.mode === "converse") {			  
-            node.status({fill:"blue", shape:"dot", text:"Starting Dialog Conversation"});
```
Comment (yrezgui) [inline services/dialog/v1.js:150]: Codacy Issue found: [Expected indentation of 8 space characters but found 10.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2462016486/issues/source?bid=3225115&fileBranchId=3255594#l149)

**Comment by chughts** [comment]
Pull request merged as requested, as it had no conflicts.

**Pull request merged**